src/vaadin-date-picker-mixin.html

```diff
@@ -399,6 +409,7 @@
       });
       this.addEventListener('keydown', this._onKeydown.bind(this));
       this.addEventListener('input', this._onUserInput.bind(this));
+      this._inputElement.addEventListener('input', this.__onUserInput.bind(this));
```

Comment:
Would it work if we reused the existing input listener above:

```diff
- this.addEventListener('input', this._onUserInput.bind(this));
+ this.addEventListener('input', this.__onUserInput.bind(this));
```

Reply:
No, `should fire the event on input value presence change` this fails

Reply:
I guess it is because the current `CustomEvent('input')` event we create in the test doesn't bubble although it should because the native `input` event bubbles. 

```js
dispatchEvent(new CustomEvent('input', { bubbles: true }); 
```

Reply:
According to W3C specs, it should also be composed:

```js
input.dispatchEvent(new CustomEvent('input', { bubbles: true, composed: true }));
```

Let me know if it will work out.

Reply:
No, it still doesn't